### PR TITLE
Make Path and .path() expand users and environment variables.

### LINKED ISSUE #3: `fields.Path` doesn't handle home directories
In https://bugs.launchpad.net/zope.configuration/+bug/494493, @batlock666 reported:

> The fieldtype Path from zope.configuration.fields doesn't handle home directories, indicated by a ~ in the path. The path ~/some.file will be interpreted as /path/to/some/package/~/some.file, while it should be something like /home/someuser/some.file.

@freddrake followed up:

> How much need is there for ~/some.file or ~user/some.file in ZCML?
> 
> If you're not using this from ZCML, could you describe you use case?

@batlock666 replied:

> I use it in a custom ZCML-directive. The directive defines a configuration-file in INI-format. The handler for the directive reads the file, and makes the contents available in a global utility:
> 
> In ZCML, I define:
> 
> ``` xml
> <ugent:configurationFile path="~/some.config.ini"
>                          name="some_config" />
> ```
> 
> Then in Zope, I can do this:
> 
> ``` python
> config = getUtility(IConfigurationFile, name="some_config")
> print config.getSections()
> ['some_section', 'some_other_section']
> ```
> 
> I use this to define connection-settings for MySQL and LDAP.

### PR DESCRIPTION
**Pull request by jamadden**
Fixes #3

### CODE REVIEW

**Comment by jamadden** [comment]
Thanks!

File: src/zope/configuration/tests/test_config.py [hunk 1]
```diff
@@ -163,6 +163,31 @@ class stub:
...
+        # HOME should be absolute
+        os.environ['HOME'] = os.path.join(os.sep, 'HOME')
+        try:
+            path = c.path('~')
+            self.assertEqual(path, os.path.join(os.sep, 'HOME'))
+        finally: # pragma: no cover
+            if old_home:
+                os.environ['HOME'] = old_home
+            else:
+                del os.environ['HOME']
```
Comment (mgedmin) [inline src/zope/configuration/tests/test_config.py]: Drive-by remark: this sort of thing would be much nicer with `mock.patch.dict()`, if we added mock as a dependency.
